### ISSUE TITLE
Using Munki to deploy MunkiReport client & Python 2.7

### ISSUE THREAD

**Issue by jelockwood** [munkireport/munkireport-php#1487]
MunkiReport on a client Mac still requires Python 2.7 to run the Python scripts that collect information for the majority of MunkiReport modules.

As of macOS 12.3 macOS no longer includes _any_ Python. The author of MunkiReport has kindly provided a pre-built installer for Python 2.7

https://github.com/munkireport/MunkiReport-Python

This is _supposed_ to install a separate copy of Python in to /Library/MunkiReport and as such should not conflict with any other copies of Python.

I have added the MunkiReport client installer to Munki, and also added the Python 2.7 installer to Munki. I have configured Munki to 'require' that the Python installer be run _before_ installing the MunkiReport installer.

Unfortunately under both macOS Big Sur and macOS Monterey I find that Munki fails to install the Python module and produces the following error.

<img width="1112" alt="Screenshot 2022-09-21 at 10 58 46" src="https://user-images.githubusercontent.com/4300786/191478019-8279d36d-fc0f-4ba8-b64b-f401477169d6.png">

Even though macOS Big Sur will still have a built-in copy of Python it will be in a different location and should not conflict. The test macOS Monterey machine is on 12.6 so does not have _any_ Python installed as standard but Munki _itself_ has it's own bundled copy but again that should be in a different location.

The authors own Wiki documentation recognises that deploying MunkiReport via Munki is a common scenario.

My best guess is that Munki aka Managed Software Centre is running it's own Python and even though the installer package for the 2.7 version will be installing in to a different location Munki has defined the _name_ of Python.app as a blocking app but not the location. (I have confirmed this is true in the Munki pkginfo file for the Python 2.7 installer.)

Other than removing this blocking app entry from the pkginfo - and potentially it will be truely blocking if/when the author issues an updated version of Python 2.7 does anyone have any recommendations for a better approach?

**Comment by tuxudo** [comment]
Verify that you are trying to install the MunkiReport version of Python 2. 

Do not have Python 2 set as required for MunkiReport, but as a managed install. 

The MunkiReport Python 2 package should not have anything set in the blocking applications array.

**Comment by jelockwood** [comment]
When I added it to Munki it automatically added the blocking entry and normally this would make sense and be desirable. I removed it and the install succeeded and unlike say updating Word I think in the case of Python updating a running copy will have less impact.

**Comment by SujeevPachala** [comment]
Can someone post what needs to be done if I encounter this issue?
Do we have to push Python2 as auto install app via Munki?

Sujeev

**Comment by jelockwood** [comment]
@SujeevPachala 
I had created my own AutoPkg recipe to add the Python2 installer to my Munki repo from @tuxudo repo here on GitHub. This had _automatically_ included a 'blocking app' entry for Python.app 

As described this actually found a running Python.app belonging to Munki and not MunkiReport, which is what caused the problem.

I solved the issue by editing the pkginfo file for Python2 in my Munki repo to remove this blocking app definition. This was the only change I made. I still left things set so that my MunkiReport pkginfo stated that the Python2 installer was required and this means Munki automatically installs Python2 first and then the MunkiReport client. (As I wanted.)

I have both the MunkiReport and the Python2 installers set in Munki as being unattended installs and managed installs. I want to ensure they are automatically installed on every Mac. (This is all now working fine and I have hundreds of Macs reporting to the MunkiReport server.)

My next project requires Python3 on every Mac so I will have to do the same for that one as well.
